File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
        int longestConsecutive(vector<int>& nums) {
        unordered_set<int> set (nums.begin(), nums.end());
        int max_len = 0, len = 0;
        for(int num : set)
        {
            if(set.count(num - 1) == 0) // does not exist
            {
                for(int cur_num = num; set.count(cur_num) != 0; cur_num++)
                {
                    len++;
                }
                max_len = max(max_len, len);
                len = 0;
            }
        }
        return max_len;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if (len(nums) == 0): return 0
        out = 0
        nums_set = set(nums)

        for num in nums_set:
            if (num - 1) not in nums_set:
                length = 1
                while (num + length) in nums_set:
                    length += 1
                out = max(out, length)
        return out